aten/src/ATen/core/ivalue_inl.h

```diff
@@ -947,9 +959,13 @@ struct C10_EXPORT ivalue::Object final : c10::intrusive_ptr_target {
 
   c10::intrusive_ptr<Object> deepcopy(IValue::HashAliasedIValueMap& memo) const;
 
+  bool is_weak_compilation_ref() const {
```

Comment:
Continuing on the theme of "I am scared of mixed ownership"—maybe we should make all these `weak_compilation_ref` methods named something much more specific, to discourage their generic usage. Like: `is_frozen_constant`. That way you really have to be aiming hard to shoot yourself in the foot + it would make some of the constant prop code clearer.

Reply:
it doesnt occur today but in theory we will be able to make constants of objects in the future

Reply:
yeah, my pref would be to at least name it as forbiddingly as possible for now, in lieu of more intrusive changes for safety. So if we add constants of objects in the future we may be able to loosen the naming.

Reply:
i have to do more changes when i fix deserialization so will bucket more api changes/etc then,  have updated some of the  apis here and added comments